### ISSUE TITLE
skaffold init fails to identify proper workspace for artifact

### ISSUE THREAD

**Issue by nkubala** [GoogleContainerTools/skaffold#3760]
```
➜  getting-started git:(master) ✗ tree
.
├── README.md
└── app
    ├── deploy
    │   ├── Dockerfile
    │   └── k8s-pod.yaml
    └── main.go

2 directories, 4 files

➜  getting-started git:(master) ✗ skaffold init
apiVersion: skaffold/v2alpha4
kind: Config
metadata:
  name: getting-started
build:
  artifacts:
  - image: skaffold-example
    context: app/deploy
deploy:
  kubectl:
    manifests:
    - app/deploy/k8s-pod.yaml

Do you want to write this configuration to skaffold.yaml? [y/n]: y
Configuration skaffold.yaml was written
You can now run [skaffold build] to build the artifacts
or [skaffold run] to build and deploy
or [skaffold dev] to enter development mode, with auto-redeploy

➜  getting-started git:(master) ✗ skaffold dev
Listing files to watch...
 - skaffold-example
FATA[0000] watching files for artifact skaffold-example: listing files: listing files: file pattern [main.go] must match at least one file
```

the `context` generated in the build config is `app/deploy`, which is where it found the **builder**, but it should actually be `app/`.

**Comment by balopat** [comment]
What do you think we should do? 
I think: 

Interactive mode: we should ask interactively the user for other build settings (build context, build args). 
Non-interactive mode: only support Dockerfile's at the root of the context.

**Comment by nkubala** [comment]
how do we know where the root of the context is though? we can't scan every file we see in the project and know if it's "part of the context", unless we computed the build dependencies **first**, and that starts to get really hairy....

in interactive mode, I think it makes sense to optionally prompt the user if they want to change basic things about the build/deploy configuration. I wonder if there would be an easy way to show the user the skaffold.yaml we're about to generate, and allow them to make quick changes to stanzas before we write it? I guess that's no different than just writing it and letting them edit later though.

i'm starting to think "best effort" is ok here. i'm not sure how we could be smarter about detecting project root vs. build context root without getting really deep into the build dependency weeds. maybe we could just document this more explicitly in the init docs, and also update the error message we provide to say something like 

`FATA[0000] watching files for artifact skaffold-example: listing files: listing files: file pattern [main.go] must match at least one file. is your build context configured correctly?`

**Comment by briandealwis** [comment]
@nkubala where did the `main.go` come from?  The Dockerfile?  We could process the ADD/COPY references from the Dockerfile and warn if they aren't satisfied, but many Dockerfiles seem to use `COPY . <dir>`.

**Comment by nkubala** [comment]
@briandealwis yeah the file patterns are extracted from the Dockerfile ADD/COPYs. what you suggested is actually exactly what we do at runtime during the build, but I suppose trying to be smart and matching source files at init runtime could work? in fact I think `COPY . <dir>` could be useful for us since `<dir>` is probably exactly what we want to use for the generated build context in the skaffold.yaml? 🤷
